src/aiohappyeyeballs/_staggered.py

```diff
@@ -156,9 +156,9 @@ async def run_one_coro(
                 # so we have no winner and all coroutines failed.
                 break
 
-            while tasks:
+            while tasks or start_next:
```

Comment:
If `start_next` and `tasks` all finished at the same time we could end up with another timer handle running that never got cancelled